File: src/recognition.js
'use strict';

const request = require('request');

// Replace <Subscription Key> with your valid subscription key.
const subscriptionKey = 'fabb4183fc89457f8e0a19eda2d04347';

// You must use the same location in your REST call as you used to get your
// subscription keys. For example, if you got your subscription keys from
// westus, replace "westcentralus" in the URL below with "westus".
const uriBase = 'https://australiaeast.api.cognitive.microsoft.com/face/v1.0/detect';

const imageUrl = 'https://i.kinja-img.com/gawker-media/image/upload/s--rKYOx8vT--/c_scale,f_auto,fl_progressive,q_80,w_800/19cmhfu7ebpy7jpg.jpg';

// Request parameters.
const params = {
    'returnFaceId': 'true',
    'returnFaceLandmarks': 'false',
    'returnFaceAttributes': 'age,gender,headPose,smile,facialHair,glasses,' +
        'emotion,hair,makeup,occlusion,accessories,blur,exposure,noise'
};

const options = {
    uri: uriBase,
    qs: params,
    body: '{"url": ' + '"' + imageUrl + '"}',
    headers: {
        'Content-Type': 'application/json',
        'Ocp-Apim-Subscription-Key' : subscriptionKey
    }
};

export default request.post(options, (error, response, body) => {
  if (error) {
    console.log('Error: ', error);
    return;
  }
  let jsonResponse = JSON.stringify(JSON.parse(body), null, '  ');
  console.log('JSON Response\n');
  return jsonResponse
});

File: backend/recognition.py
import requests
import json

subscription_key = 'fabb4183fc89457f8e0a19eda2d04347'
assert subscription_key

face_api_url = 'https://australiaeast.api.cognitive.microsoft.com/face/v1.0/detect'

image_url = 'https://i.kinja-img.com/gawker-media/image/upload/s--rKYOx8vT--/c_scale,f_auto,fl_progressive,q_80,w_800/19cmhfu7ebpy7jpg.jpg'

headers = { 'Ocp-Apim-Subscription-Key': subscription_key }

params = {
    'returnFaceId': 'true',
    'returnFaceLandmarks': 'false',
    'returnFaceAttributes': 'age,gender,headPose,smile,facialHair,glasses,emotion,hair,makeup,occlusion,accessories,blur,exposure,noise',
}

response = requests.post(face_api_url, params=params, headers=headers, json={"url": image_url})
print(json.dumps(response.json()))